src/components/timeline/form/TimelineEditorPanel.svelte

```diff
@@ -312,25 +312,29 @@
             ...l,
             filter: {
               ...l.filter,
-              ...(l.filter.activity
-                ? {
-                    ...l.filter.activity,
-                    types: values,
-                  }
-                : {}),
+              activity: {
```

Comment:
wait why didn't TS catch this? also if we do it this way, there's a chance that `activity` is just an empty object. is that a correct state?

Reply:
Just a guess here, but I think it did not catch it because the `activity` and `resource` keys can both be `undefined` so just spreading an object is technically valid (although it shouldn't be).
Updated the code to hopefully make it more clear.

Reply:
That's a little worrisome. I wonder if it's a setting or something.